Makefile

```diff
@@ -41,12 +41,13 @@ nag:
 	cfn_nag_scan --input-path code/solutions --ignore-fatal
 
 # Versioning and releases
+.PHONY: version release
 version: $(VENV_NAME)
 	@$(VENV_NAME)/bin/bumpversion $(part)
 
-release:
+release: # run on main branch only
 	@TAG_VERSION=$(shell bumpversion --dry-run --list .bumpversion.cfg | grep current_version | sed s/'^.*='//); \
-		git push origin "v$${TAG_VERSION}" && git push
+		git tag -a "v$${TAG_VERSION}" -m "" && git push origin "v$${TAG_VERSION}"
```

Comment:
For my own curiousity, why are annotated tags used instead of lightweight tags?

Reply:
So this is my personal opinion, and I may be completly wrong here, bot for me:

I use **annotated** tags for releases, just to give it it the little bit more matadata like who/what/why and marked them release ready, if that makes sense.

Reply:
The **lightweight** tags, I don't use much, sometimes when i want to test something and find it later, I may tag it. So mainly I use lightweight tags for private use.

Reply:
Useful to know. Thanks Franco!